src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -621,19 +616,14 @@ internal virtual T ReadAsPropertyNameCore(ref Utf8JsonReader reader, Type typeTo
         /// <remarks>Method should be overridden in custom converters of types used in serialized dictionary keys.</remarks>
         public virtual void WriteAsPropertyName(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
         {
-            if (!IsInternalConverter &&
-                options.SerializerContext is null && // For consistency do not return any default converters for
-                                                     // options instances linked to a JsonSerializerContext,
-                                                     // even if the default converters might have been rooted.
-                DefaultJsonTypeInfoResolver.TryGetDefaultSimpleConverter(TypeToConvert, out JsonConverter? defaultConverter))
+            // .NET 5 backward compatibility: hardcode the default converter for primitive key serialization.
```

Comment:
nit on comment: this applies more to non-primitive keys no?

Reply:
"primitive" is probably abuse of terminology, but generally speaking it's converters that serialize as JSON primitives: string, number, boolean.

Reply:
Oh nvm this is applicable to _custom_ primitive converters the don't override this method.